core/src/kura.rs

```diff
@@ -328,6 +347,35 @@ impl Kura {
         data.pop();
         data.push((block.hash(), Some(block)));
     }
+
+    /// Store WSV to disk.
+    pub fn store_wsv(&self, wsv: &WorldStateView) {
+        let data: WSVData = wsv.into();
+        let path = self.block_store_path.join(WSV_FILE_NAME);
+
+        std::thread::spawn(move || {
+            let bin_data: Vec<u8> = data.encode();
+
+            let mut wsv_file = std::fs::OpenOptions::new()
+                .write(true)
+                .truncate(true)
+                .create(true)
+                .open(path.clone())
+                .expect("Failed to open wsv file");
+
+            wsv_file
+                .write_all(&bin_data)
+                .expect("Failed to write wsv file");
+        });
+    }
+
+    /// Load WSVData from disk.
+    pub fn try_load_wsv(&self) -> Option<WSVData> {
```

Comment:
I'd use explicit errors. If it seems like to much unrelated works consider creating a new task please

Reply:
`Option` is fine, if the error that you return has context only when forwarded up the stack.